### PR TITLE
Check if the found nodejs version supports the required Node-API version

### PR DESCRIPTION
**Pull request by hikinggrass**
Signed-off-by: Kai-Uwe Hermann <kai-uwe.hermann@pionix.de>

### CODE REVIEW

File: everestjs/CMakeLists.txt [hunk 1]
```diff
@@ -7,13 +7,65 @@ target_link_libraries(everestjs
...
+        NODE
+        node
+        REQUIRED
+)
+
+# check node API version based on https://nodejs.org/api/n-api.html#node-api-version-matrix
+execute_process(COMMAND ${NODE} --version
+        OUTPUT_VARIABLE NODE_VERSION
+)
+string(REGEX REPLACE "[\r\n\"]" "" NODE_VERSION "${NODE_VERSION}")
```
Comment (a-w50) [inline everestjs/CMakeLists.txt]: would be OUTPUT_STRIP_TRAILING_WHITESPACE enough in the execute_process cmd?

**Review by a-w50 — COMMENTED** [review]
I would probably put this additional logic into a separate cmake "module" and include it here
Otherwise, well done

File: everestjs/CMakeLists.txt (hunk 1 above)
Comment (hikinggrass) [reply everestjs/CMakeLists.txt]: probably, good idea

**Review by a-w50 — APPROVED** [review]
Fine

**Pull request merged**